src/lib/storage/proposal/partition_creator.rb

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "fileutils"
+require "storage/planned_volumes_collection"
+require "storage/disk_size"
+require "storage/proposal/refined_devicegraph"
+require "storage/proposal/exceptions"
+
+module Yast
+  module Storage
+    module Proposal
+      using RefinedDevicegraph
+
+      # Class to create partitions in the free space detected or freed by the
+      # SpaceMaker.
+      class PartitionCreator
+        include Yast::Logger
+
+        attr_accessor :settings
+
+        VOLUME_GROUP_SYSTEM = "system"
+        FIRST_LOGICAL_PARTITION_NUMBER = 5 # Number of the first logical partition (/dev/sdx5)
+
+        # Initialize.
+        #
+        # @param original_graph [::Storage::Devicegraph] initial devicegraph
+        # @param settings [Proposal::Settings] proposal settings
+        def initialize(original_graph, settings)
+          @original_graph = original_graph
+          @settings = settings
+        end
+
+        # Returns a copy of the original devicegraph in which all the needed
+        # partitions have been created.
+        #
+        # @param volumes [PlannedVolumesCollection] volumes to create
+        # @param target_size [Symbol] :desired or :min
+        # @return [::Storage::Devicegraph]
+        def run(volumes, target_size)
```

Comment:
"run" is a bit too generic for my taste. I think we should come up with a better name.